plugins/plugin-bitbucket/codenvy-plugin-bitbucket-pullrequest/src/main/java/com/codenvy/plugin/pullrequest/client/BitbucketHostingService.java

```diff
@@ -146,7 +151,9 @@ public boolean isHostRemoteUrl(final String remoteUrl) {
                                                  final BitbucketPullRequestLocation source = pullRequest.getSource();
                                                  if (author != null && source != null) {
                                                      final BitbucketPullRequestBranch branch = source.getBranch();
-                                                     if (username.equals(author.getUsername()) && branchName.equals(branch.getName())) {
+                                                     //Bitbucket Server adds '~' to authenticated user, need to substring it.
+                                                     String userName = username.startsWith("~") ? username.substring(1) : username;
```

Comment:
It's hard to say that it is good practice to have userName and username variables